pkg/v1/tkr/controllers/source/tkr_source_controller.go

```diff
@@ -50,19 +50,19 @@ type reconciler struct {
 }
 
 // Reconcile performs the reconciliation step
-func (r *reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	ctx, cancel := context.WithCancel(r.ctx)
+func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctxCancel, cancel := context.WithCancel(r.ctx)
```

Comment:
Just noticed this: `ctx` should be used as the parent for `ctxCancel`. We've been using `r.ctx` for the lack of a context coming with the request. Now that we have it, `ctx` should be used.

The expectation of the caller is: if/when the passed context `ctx` expires (e.g. has timed out or been canceled), the call should error out. 

We don't *need* our own context here, can just use the passed `ctx`, but it's not a bad idea to have one. 

The reason, why the original `ctx` was created here, was: we needed a context to pass to `r.client` calls, but nothing was coming along with the request. Passing `r.ctx` seemed wrong as it's scoped to the lifetime of the whole reconciler, so we created our own request context.

Reply:
Looks good now, thanks!